TPP.Core/Commands/Definitions/ManagePollCommands.cs

```diff
@@ -0,0 +1,54 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using TPP.ArgsParsing.Types;
+using TPP.Persistence.Models;
+using TPP.Persistence.Repos;
+
+namespace TPP.Core.Commands.Definitions
+{
+    class ManagePollCommands : ICommandCollection
+    {
+        private static readonly Role[] AllowedRoles = { Role.Moderator, Role.Operator };
+
+        public IEnumerable<Command> Commands => new[]
+        {
+            new Command("createpoll", StartPoll)
+            {
+                Aliases = new[] { "startpoll" },
+                Description = "Starts a poll with single choice. " +
+                              "Argument: <PollName> <PollCode> <MultipleChoice> <AllowChangeVote> <Option1> <Option2> <OptionX> (optional). " +
+                              "Underscores in the poll name will be replaces with spaces."
+            },
+        }.Select(cmd => cmd.WithCondition(
+            canExecute: ctx => ctx.Message.User.Roles.Intersect(AllowedRoles).Any(),
+            ersatzResult: new CommandResult { Response = "Only moderators can manage polls" }));
+
+        private readonly IPollRepo _pollRepo;
+
+        public ManagePollCommands(IPollRepo pollRepo)
+        {
+            _pollRepo = pollRepo;
+        }
+
+        public async Task<CommandResult> StartPoll(CommandContext context)
+        {
+            (string pollName, string pollCode, bool multiChoice, bool allowChangeVote, ManyOf<string> options) =
+                await context.ParseArgs<string, string, bool, bool, ManyOf<string>>();
```

Comment:
imo multiChoice and allowChangeVote should be optional and should have a default value of false, if the bool parser can't parse params 3 and 4 into a bool they should be grouped into options, may require a change to args parser for that though

Reply:
sounds unnecessarily complicated to me since creating polls is a rare and administrative process anyway

Reply:
that's fair I guess, not really necessary